src/Infrastructure/Stize.Infrastructure.Azure/Sql/SqlDatabaseExtensions.cs

```diff
@@ -67,36 +66,140 @@ public static SqlDatabaseBuilder ElasticPoolId(this SqlDatabaseBuilder builder,
+        /// Restores a deleted database using the Restorable Dropped Database Resource ID.
         /// </summary>
         /// <param name="builder"></param>
-        /// <param name="edition"></param>
+        /// <param name="databaseId">Resource ID of the Restorable Dropped Database to restore from</param>
+        /// <param name="pointInTime">Point in time that you want to restore from. Only specify to restore from an earlier point in time.</param>
         /// <returns></returns>
-        public static SqlDatabaseBuilder SkuTier(this SqlDatabaseBuilder builder, Input<string> edition)
+        public static SqlDatabaseBuilder SetAsRestore(this SqlDatabaseBuilder builder, Input<string> databaseId, Input<string> pointInTime = null)
         {
-            builder.SkuArguments.Tier = edition;
+            /* TODO: Improve usability of the method
+             * "sourceDatabaseId must be the restorable dropped database resource ID and sourceDatabaseDeletionDate is ignored." - https://www.pulumi.com/docs/reference/pkg/azure-native/sql/database/#createmode_csharp
+             * ^ documentation states to assign SourcedDatabaseId - must use RestorableDroppedDatabaseId when using CreateMode.Restore
+             * Currently, user must pass the RestorableDroppedDatabaseId. This isn't friendly though as they would need to know the id.
+             * More friendly for user to input a db name, server name, and rg name (and optional point in time to restore for more earlier backups)
+             * It is possible to access a dropped (deleted) database using PS command Get-AzureRmSqlDeletedDatabaseBackup and providing parameters (resourcegroup, server, database)
+             * There is a GetDatabase function provided by Pulumi, but it doesn't seem to be able to grab deleted databases. and there is no GetDeletedDatabase function.
+            */
+            builder.Arguments.CreateMode = CreateMode.Restore;
+            builder.Arguments.RestorableDroppedDatabaseId = databaseId;
+            builder.Arguments.RestorePointInTime = pointInTime ?? null;
             return builder;
         }
 
         /// <summary>
-        /// Restores the database from an existing one
+        /// Restores a deleted database using the database's original resource ID.
         /// </summary>
         /// <param name="builder"></param>
-        /// <param name="databaseId"></param>
+        /// <param name="databaseId">Original Resource ID of the database to restore from.</param>
+        /// <param name="deletionDate">Deletion date of the database in (ISO 8601 format).</param>
+        /// <param name="pointInTime">Point in time that you want to restore from (ISO 8601 format). Only specify to restore from an earlier point in time.</param>
+        /// <returns></returns>
+        public static SqlDatabaseBuilder SetAsRestore(this SqlDatabaseBuilder builder, Input<string> databaseId, Input<string> deletionDate, 
```

Comment:
Should we rename this to "CreateAsRestoreOf"? I feel the current name is a bit confusing.